src/realm/util/thread.cpp

```diff
@@ -95,6 +95,25 @@ void Thread::set_name(const std::string& name)
     int r = pthread_setname_np(name.data());
     if (REALM_UNLIKELY(r != 0))
         throw std::system_error(r, std::system_category(), "pthread_setname_np() failed");
+#elif defined(_WIN32)
+    // SetThreadDescription was only introduced in Windows 10 1607, so we need to dynamically load it.
+    HRESULT(WINAPI * SETTHREADDESCRIPTION)(HANDLE hThread, PCWSTR threadDescription) = nullptr;
```

Comment:
Consider naming this just `SetThreadDescription` and using `&::SetThreadDescription to get the pointer below. Or just move this into the `#if` and delete the `#else` so that it just uses the function directly.